src/include/migraphx/matcher.hpp

```diff
@@ -535,6 +535,19 @@ auto same_shape(Ms... ms)
     return all_of(same_shape(ms)...);
 }
 
+template <class T>
+inline auto has_value(T x, float tolerance = 1e-6)
+{
+    return make_basic_pred_matcher([=](instruction_ref ins) {
+        auto l = ins->get_literal();
+        if(l.empty())
```

Comment:
This should check that there is only one element(ie `ins->get_shape().elements() == 1`) since you are only checking the first element.